compile.sh:increase MAXCORES

andy5995 opened the pull request:
This increases MAXCORES so when building with [distcc](https://www.distcc.org/) the benefit is significantly improved.

It would probably be better just to remove the limitation completely. What do you think?

tothxa reviewed (commented):
It was said before that `compile.sh` is aimed at non-developers.

If you really need this, I think a better solution would be to use `$MAXCORES` from the environment if set.

tothxa commented:
OK, so this was just a first draft.

I thought some more about it, and instead of setting `$MAXCORES` it would be more straightforward to add something like a `-J`/`--force-cores` option that wouldn't be checked against `$MAXCORES`.

-----

On the idea of distcc: I guess it may be very useful, but probably could be helped by making it possible to explicitly tell the script about it. But then again, that's probably too much for a specialised use case for more advanced usage.

There's also the problem that distcc only distributes the C++ compilation tasks. Of course that's by far the longest (except for final linking and stripping of the executable, which can't be parallelised :((( ), but it's probably not desirable to swamp the local machine with a gazillion of translation compile tasks. Like I already skip building the tests because running them in parallel needs more RAM than I have, and I don't want to turn on swap on a SSD.

So you're better off if you just run cmake separately, so then you can use `make`/`ninja` `-j 100 widelands` with distcc, and `-j $MAXCORES lang` for translations, etc.

andy5995 commented:
@tothxa I think I understood all your points. Thanks for explaining. I pushed a change that may address all the issues. What do you think? Or would you still prefer a "--force" option?

tothxa reviewed (commented):
Thanks for addressing my concerns.

Let's wait for other comments though.

hessenfarmer commented:
@tothxa @Noordfrees @andy5995 
sorry I can't be of any help here as I don't understand much from the compile script. I am still a windows guy

andy5995 committed:
compile.sh:allow user to set number of cores

This allows a user to override the the limit of max cores detected.
However, it won't try to use more than max cores when building
translations.

tothxa reviewed (commented):
We still need to check all possible targets, and separate C++ compilation from the rest, but I certainly wouldn't be able to review all cmake files to do this.

What I know to exist:

 - codecheck (python): enabled (by cmake) with debug, but also has its own cmake option. It's independent of compilation, but it is better run before it, like in the github actions. It can't be run if not enabled, but normally runs with compilation if enabled. And `compile.sh` completely ignores it currently, doesn't even have an idea whether it's enabled.
 - tests: I think it involves some compilation but also running the testsuites afterwards, and I don't know whether these can be easily separated (compilation can be parallel with main compilation, while running must come afterwards)
 - website tools: compilation, can be parallel

andy5995 commented:
It all seems like a lot of extra code just to alter the behavior of an option as standard as '-j'. Working on it doesn't interest me at all at this time.
